frontend/src/components/ButtonComponent.vue

```diff
@@ -1,13 +1,39 @@
 <!-- Button to interact with -->
 <template>
-  <InteractableComponent @click="(e) => $emit('click', e)">
-    <!-- Slot for the text of the button -->
-    <slot></slot>
-  </InteractableComponent>
+  <div>
+    <div class="relative">
+      <div class="absolute w-full h-full text-center" v-if="loading">
+        <div
+          class="absolute h-full w-full bg-gray-300 opacity-70 rounded-md"
+        ></div>
+        <FontAwesomeIcon icon="spinner" size="1x" class="animate-spin mt-2" />
+      </div>
+      <InteractableComponent @click="(e) => $emit('click', e)">
```

Comment:
We could add some functionality that prevents click events while the spinner is showing - this way we might avoid accidential double-presses or similar - Opinions?

Reply:
either this or just implement not calling the coresponding method in the view twive

Reply:
I feel like disabling while spinning is the much easier approach - not calling twice requires checking timings between presses, etc.